### ISSUE TITLE
Rule 'always-return' fails on import

### ISSUE THREAD

**Issue by ascariandrea** [xjamundx/eslint-plugin-promise#36]
Not sure what is the bug here, but I got this:

```
0:0  error  Each then() should return a value or throw  promise/always-return
```

and at the `0 line` I have an import declaration:

``` js
import API from 'js/util/api';
```

The weird thing here is the I have this error only in one file.

**Comment by xjamundx** [comment]
that's super weird. anything else you can share about why it's happening?

**Comment by ascariandrea** [comment]
Ok, I got it.
With `eslint@2.13.1` the `loc: SourceLocation` argument passed to `context.report()` is treated [differently](https://github.com/eslint/eslint/blob/2.x/lib/rule-context.js#L130) from the location get from `node` (it's equal for `eslint@^3, but it works)`.

So, for `eslint@^2` the `context.report({...})` should be 
```js
context.report({
 message: 'Each then() should return a value or throw',
 loc: branch.loc.start
})
```

Upgrading `eslint` up to `3` version solved the problem, but please fix by replacing `loc` with `node` as it seems the preferred way.

**Comment by xjamundx** [comment]
@ascariandrea great research!

**Comment by ascariandrea** [comment]
Do you want me to submit a PR?

**Comment by xjamundx** [comment]
Yes please submit a fix if you have it
